src/System.Private.ServiceModel/src/System/ServiceModel/Dispatcher/XmlSerializerOperationFormatter.cs

```diff
@@ -20,13 +20,20 @@ internal class XmlSerializerOperationFormatter : OperationFormatter
         private const string soap11Encoding = "http://schemas.xmlsoap.org/soap/encoding/";
         private const string soap12Encoding = "http://www.w3.org/2003/05/soap-encoding";
 
+        private bool isEncoded;
         private MessageInfo _requestMessageInfo;
         private MessageInfo _replyMessageInfo;
 
         public XmlSerializerOperationFormatter(OperationDescription description, XmlSerializerFormatAttribute xmlSerializerFormatAttribute,
             MessageInfo requestMessageInfo, MessageInfo replyMessageInfo) :
             base(description, xmlSerializerFormatAttribute.Style == OperationFormatStyle.Rpc, xmlSerializerFormatAttribute.IsEncoded)
         {
+            if (xmlSerializerFormatAttribute.IsEncoded && xmlSerializerFormatAttribute.Style != OperationFormatStyle.Rpc)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.SFxDocEncodedNotSupported, description.Name)));
+            }
+            this.isEncoded = xmlSerializerFormatAttribute.IsEncoded;
```

Comment:
A few nits.
1. Blank line after closing brace.
2. Remote this. qualifier

Reply:
Done